fyle_rest_auth/authentication.py

```diff
@@ -64,13 +66,32 @@ def validate_token(access_token_string: str) -> Dict:
 
             api_headers = {'Authorization': '{0}'.format(access_token_string)}
 
-            response = requests.get(my_profile_uri, headers=api_headers)
+            email_unique_key = 'email_' + unique_key_generator[2]
+            user_unique_key = 'user_' + unique_key_generator[2]
 
-            if response.status_code == 200:
-                result = json.loads(response.text)['data']
+            email = cache.get(email_unique_key)
+            user = cache.get(user_unique_key)
+
+            if email is None:
+                if user is None:
+                    cache.delete_many([email_unique_key, user_unique_key])
+                    response = requests.get(my_profile_uri, headers=api_headers)
+
+                    if response.status_code == 200:
+                        result = json.loads(response.text)['data']
+
+                        cache.set(email_unique_key, result['employee_email'], 900)
```

Comment:
The value of cache expiry should come from the Django settings file, in the project it is imported